### PR TITLE
Upgrade to webpack and eslint

### LINKED ISSUE #68: issue with webpack and react-ga/dist/react-ga.js
When I `require('react-ga')` in a webpack-bundled app, I get the following warning:

```
WARNING in ./~/react-ga/dist/react-ga.js
Critical dependencies:
1:480-487 This seems to be a pre-built javascript file. Though this is possible, it's not recommended. Try to require the original source to get better results.
 @ ./~/react-ga/dist/react-ga.js 1:480-487
```

I'm not sure why webpack is importing the dist version, when react-ga's package.json specifies main:

```
  "main": "src/index.js",
```

But the real problem is that this dist version results in an odd error in the browser:

```
Uncaught TypeError: Cannot read property 'createClass' of undefined
```

That error is triggered by this line:
https://github.com/react-ga/react-ga/blob/master/dist/react-ga.js#L44

```
var React = (typeof window !== "undefined" ? window['React'] : typeof global !== "undefined" ? global['React'] : null);
```

which seems to be making some odd assumptions about how to access React on window or global. Not sure what's going on there, but the result for me is `null` because React isn't exposed as a global in my app.

It's possible that this dist file isn't meant to be used in bundled apps, so maybe the problem is with webpack or my webpack config. Not sure how to suss that out..

As an interim solution, I've added this to the `resolve.alias` section of my webpack config:

```
resolve: {
    alias: {
        // fix for React.createClass bug in react-ga
        'react-ga': 'react-ga/src/index.js',
    }
}
```

Anyone else experiencing this issue?

### LINKED ISSUE #77: Upgrade to Webpack, Babel, ESLint
It's probably good to upgrade this project to use the common React ecosystem tools 😄

### PR DESCRIPTION
**Pull request by SimeonC**
Fixes #68
Fixes #77

Hi all, pretty big PR here to get our stack up to scratch. There should be 0 functionality changes and only one test did I have to remove as it seemed a little bit redundant - also I couldn't actually get the test to work either.

EDIT: If anyone wants to test they can use `npm i github:SimeonC/react-ga#webpack-upgrade-version` which has the built files in it.

### CODE REVIEW

**Comment by SimeonC** [comment]
Hmmm, I think I have to update the node version we're using... Seems a lot of the plugins latest versions don't work on Node 0.12.

**Comment by SimeonC** [comment]
Tests passed. I'd like to merge this in but I'd need a Review.

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    'react-ga': './src/index.js',
+    'react-ga.min': './src/index.js'
+  },
+  output: {
+    path: path.resolve('./dist'),
+    filename: '[name].js'
+  },
+  module: {
+    loaders: [
+      { test: /\.js$/, loader: 'babel-loader', exclude: /node_modules/ },
```
Comment (yangshun) [inline webpack.config.js]: These two lines can be combined into one using `test: /\.jsx?$/`

**Comment by yangshun** [comment]
@SimeonC Could you create an account on npmjs and let me know your username? I'll add you into it after you do so so that you can publish it.

File: webpack.config.js (hunk 1 above)
Comment (SimeonC) [reply webpack.config.js]: Thanks for catching that! I'll fix before the merge.

My npmjs username is "simeonc" https://www.npmjs.com/~simeonc, thanks.

**Pull request merged**